### ISSUE TITLE
Bug on getDirections function

### ISSUE THREAD

**Issue by alexjoverm** [dbaq/angularjs-google-directions#4]
I've found out that the next code line (61):

```javascript
args.unitSystem = _unitSystems[args.unitSystem] || googleMaps.UnitSystem.IMPERIAL;
```

always returned 1, no matter what. The reason is because is you pass 'metric' as a parameter, the returned value of the array is 0, so then it selects the right part of the condition. It can be solved by changing it to:

```javascript
args.unitSystem = _unitSystems[args.unitSystem] || googleMaps.UnitSystem.METRIC;
```

Also I found out that if you use the same args object for multiple calls, unitSystem was modified ( to 0 or 1). That shouldn't happen, it should be copied. So to solve also this, the whole functions would be:

```javascript
getDirections: function(args) {
    var _args = angular.copy(args);
    _args.travelMode = _travelModes[args.travelMode] || googleMaps.TravelMode.DRIVING;
    _args.unitSystem = _unitSystems[args.unitSystem] || googleMaps.UnitSystem.METRIC;
    return exec(_args);
}
```

Hope to help.

Best regards

**Comment by dbaq** [comment]
hi @alexjoverm,

Thank you so much for reporting these issues. I will dig into them really soon.

**Comment by dbaq** [comment]
Fixed via 8eaea51. Thanks for reporting.
